### PR TITLE
dategen in three phases with slurm job

### PR DESCRIPTION
**Pull request by hyjorc1**
1) Download repositories: GitHubRepoBareDownloader.java

2) Generate seq files for each project: SeqRepoGenerator.java, SeqRepoBuilder.java, run-generator.sh, slurmJob.sh

3) Combine seq files: SeqRepoCombiner.java, run-combiner.sh

### CODE REVIEW

**Comment by hridesh** [comment]
What would be the impact of this change on local data generation, e.g. on just a single machine owned by Boa user?

File: src/java/boa/datagen/slurm/GitHubRepoBareDownloader.java [hunk 1]
```diff
@@ -19,62 +20,63 @@ public static void main(String[] args) {
...
-							Thread.sleep(100);
-						} catch (InterruptedException e) {
-							e.printStackTrace();
+		INPUT_PATH = args[0];
+		OUTPUT_REPOS_PATH = args[1];
+		THREAD_NUM = Integer.parseInt(args[2]);
+
+		File input = new File(INPUT_PATH);
+
+		DownloadWorker[] workers = new DownloadWorker[THREAD_NUM];
```
Comment (hridesh) [inline src/java/boa/datagen/slurm/GitHubRepoBareDownloader.java:32]: If Boa code is intended to compile using Java 7 or higher, I would use concurrency features such as ExecutorService that is meant to provide all of these features. See https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ThreadPoolExecutor.html

File: src/java/boa/datagen/slurm/GitHubRepoBareDownloader.java [hunk 2]
```diff
@@ -19,62 +20,63 @@ public static void main(String[] args) {
...
+			if (!file.getName().endsWith(".json"))
+				continue;
+			JsonElement jsonTree = new JsonParser().parse(FileIO.readFileContents(file));
+			for (JsonElement je : jsonTree.getAsJsonArray()) {
+				String projectName = je.getAsJsonObject().get("html_url").getAsString().replace("https://github.com/",
+						"");
+
+				boolean assigned = false;
+				while (!assigned) {
+					for (int j = 0; j < THREAD_NUM; j++) {
```
Comment (hridesh) [inline src/java/boa/datagen/slurm/GitHubRepoBareDownloader.java:51]: If Boa code is intended to compile using Java 7 or higher, one could use concurrency features such as ExecutorService that is meant to provide all of these features. Just a thought. See https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ThreadPoolExecutor.html

**Comment by hyjorc1** [comment]
These changes fix and optimize the JSON retriever for extracting repo metadata.

GetReposByLanguage.java takes 4 arguments:
1. The path to the token file.
2. JSON files (100 repositories per file) output path.
3. A number to which the repository stars are greater than or equal to.
4. The programming languages (split by semicolons, e.g. "java;python") used by the repository.

**Pull request closed without merging**